cpp/client/ct.cc

```diff
@@ -404,48 +400,46 @@ static void MakeCert() {
   x->cert_info->signature->parameter = NULL;
 
   // Set the start date to now
-  X509_gmtime_adj(X509_get_notBefore(x), 0);
+  X509_gmtime_adj(X509_get_notBefore(x.get()), 0);
   // End date to now + 1 second
-  X509_gmtime_adj(X509_get_notAfter(x), 1);
+  X509_gmtime_adj(X509_get_notAfter(x.get()), 1);
 
   // Create the issuer name
-  X509_NAME* issuer = X509_NAME_new();
+  X509_NAME* issuer = X509_NAME_new();  // x takes ownership
   X509_NAME_add_entry_by_NID(issuer, NID_commonName, V_ASN1_PRINTABLESTRING,
                              const_cast<unsigned char*>(
                                  reinterpret_cast<const unsigned char*>(
                                      "Test")),
                              4, 0, -1);
-  X509_set_issuer_name(x, issuer);
+  X509_set_issuer_name(x.get(), issuer);
 
   // Create the subject name
-  X509_NAME* subject = X509_NAME_new();
-  X509_NAME_add_entry_by_NID(subject, NID_commonName, V_ASN1_PRINTABLESTRING,
-                             const_cast<unsigned char*>(
-                                 reinterpret_cast<const unsigned char*>(
-                                     "tseT")),
-                             4, 0, -1);
-  X509_set_subject_name(x, subject);
+  ScopedX509_NAME subject(X509_NAME_new());
+  X509_NAME_add_entry_by_NID(
+      subject.get(), NID_commonName, V_ASN1_PRINTABLESTRING,
+      const_cast<unsigned char*>(
+          reinterpret_cast<const unsigned char*>("tseT")),
+      4, 0, -1);
+  X509_set_subject_name(x.get(), subject.release());
 
   // Public key
-  RSA* rsa = RSA_new();
+  ScopedRSA rsa(RSA_new());
   static const unsigned char bits[1] = {3};
   rsa->n = BN_bin2bn(bits, 1, NULL);
   rsa->e = BN_bin2bn(bits, 1, NULL);
-  EVP_PKEY* evp_pkey = EVP_PKEY_new();
-  EVP_PKEY_assign_RSA(evp_pkey, rsa);
-  X509_PUBKEY_set(&X509_get_X509_PUBKEY(x), evp_pkey);
+  ScopedEVP_PKEY evp_pkey(EVP_PKEY_new());
+  EVP_PKEY_assign_RSA(evp_pkey.get(), rsa.release());
+  X509_PUBKEY_set(&X509_get_X509_PUBKEY(x), evp_pkey.release());
 
   // And finally, the proof in an extension
-  string serialized_sct_list = SCTToList(sct);
-  AddOctetExtension(x, cert_trans::NID_ctSignedCertificateTimestampList,
+  const string serialized_sct_list(SCTToList(sct));
+  AddOctetExtension(x.get(), cert_trans::NID_ctSignedCertificateTimestampList,
                     reinterpret_cast<const unsigned char*>(
                         serialized_sct_list.data()),
                     serialized_sct_list.size(), 1);
 
-  int i = i2d_X509_bio(out, x);
+  int i = i2d_X509_bio(out.get(), x.get());
```

Comment:
I think that `CHECK_GT` on the next line used to be an `assert`, hence why `i2d_X509_bio` isn't called inline. Could you make it inline while you're there?